shared/messaging/utils.go

```diff
@@ -0,0 +1,26 @@
+package messaging
+
+import (
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+// PackMessage returns a *PocketEnvelope after having packed the message supplied as an argument
+func PackMessage(message proto.Message) (*PocketEnvelope, error) {
+	anyMsg, err := anypb.New(message)
+	if err != nil {
+		return nil, err
+	}
+	return &PocketEnvelope{Content: anyMsg}, nil
+}
+
+// UnpackMessage extracts the message inside the PocketEnvelope decorating it with typing information
+func UnpackMessage(envelope *PocketEnvelope) (protoreflect.ProtoMessage, error) {
```

Comment:
Everyone who calls this function will need to return a protobuf of a specific type (as I see in the usage and in the text).

What do you think of leveraging generics?

```go
func UnpackMessage[T protoreflect.ProtoMessage] (T, error) {... }
```

Reply:
Nice catch!